OSSMDOC-582: Service Mesh 2.3  release notes (#51715)
openshift/openshift-docs: modules/ossm-rn-new-features.adoc

== New features {SMProductName} version {SMProductVersion}

> This release of {SMProductName} introduces new features, addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on OpenShift Container Platform 4.9, 4.10, and 4.11.

Reviewer: ```suggestion
This release of {SMProductName} introduces new features, addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on {product-title} 4.9, 4.10, and 4.11.
```

Author: done

=== Component versions included in {SMProductName} version {SMProductVersion}

|===
|Component |Version

|Istio
|1.14

|Envoy Proxy
|1.22.4

|Jaeger
|1.38

|Kiali
|1.57.3
|===

=== New Container Network Interface (CNI) DaemonSet container and ConfigMap

> The `openshift-operators` namespace includes a new istio CNI DaemonSet `istio-cni-node-v2-3` and a new `ConfigMap`, `istio-cni-config-v2-3`.

Author: Done, thanks for the suggestion!

When upgrading to Service Mesh Control Plane 2.3, the existing `istio-cni-node` DaemonSet is not changed, and a new `istio-cni-node-v2-3` DaemonSet is created.

This name change does not affect previous releases or any `istio-cni-node` CNI DaemonSet associated with a Service Mesh Control Plane deployed using a previous release.

=== Gateway injection support

> This release introduces Service Mesh Gateway injection. Gateway configurations are applied to standalone Envoy proxies that are running at the edge of the mesh, rather than the sidecar Envoy proxies running alongside your service workloads. This enables the ability to customize gateway options. When using gateway injection, you must create the following resources in the namespace where you want to run your gateway proxy: `Service`, `Deployment`, `Role`, and `RoleBinding`.

Reviewer 2: minor nit - "This release introduces **generally available** support for Gateway injection." It's been an Istio feature for some time now and available in previous releases, though was a noted support exception.

> This release introduces generally available support for Service Mesh Gateway injection. Gateway configurations are applied to standalone Envoy proxies that are running at the edge of the mesh, rather than the sidecar Envoy proxies running alongside your service workloads. This enables the ability to customize gateway options. When using gateway injection, you must create the following resources in the namespace where you want to run your gateway proxy: `Service`, `Deployment`, `Role`, and `RoleBinding`.

Reviewer 3: @rh-tokeefe I think that @longmuir meant to remove "Service Mesh" from the first sentence. I also suggest to remove it, because Gateway injection is a pure upstream feature, so "Service Mesh" prefix is ambiguous and may suggest that it's something OSSM-specific (what may be not appealing for users).
```suggestion
This release introduces generally available support for Gateway injection. Gateway configurations are applied to standalone Envoy proxies that are running at the edge of the mesh, rather than the sidecar Envoy proxies running alongside your service workloads. This enables the ability to customize gateway options. When using gateway injection, you must create the following resources in the namespace where you want to run your gateway proxy: `Service`, `Deployment`, `Role`, and `RoleBinding`.
```

Author: Thanks @jewertow the update has been made.

=== Istio 1.14 Support

{SMProductShortName} 2.3 is based on Istio 1.14, which brings in new features and product enhancements. While many Istio 1.14 features are supported, the following exceptions should be noted:

* ProxyConfig API is supported with the exception of the image field.
* Telemetry API is a tech preview feature.
* SPIRE runtime is not a supported feature.

=== OpenShift Service Mesh Console

> This release introduces a Developer Preview version of the OpenShift Service Mesh Console, which integrates the Kiali interface directly into the OpenShift Console. For additional information, see link:https://cloud.redhat.com/blog/introducing-the-openshift-service-mesh-console-a-developer-preview[Introducing the OpenShift Service Mesh Console (A Developer Preview)]

Author: Done, thank you for the suggestion

> This release introduces a Developer Preview version of the OpenShift Service Mesh Console, which integrates the Kiali interface directly into the OpenShift web console. For additional information, see link:https://cloud.redhat.com/blog/introducing-the-openshift-service-mesh-console-a-developer-preview[Introducing the OpenShift Service Mesh Console (A Developer Preview)]

Reviewer: ```suggestion
This release introduces a Developer Preview version of the {product-title} Service Mesh Console, which integrates the Kiali interface directly into the OpenShift web console. For additional information, see link:https://cloud.redhat.com/blog/introducing-the-openshift-service-mesh-console-a-developer-preview[Introducing the OpenShift Service Mesh Console (A Developer Preview)]
```

Author: done

=== Cluster-Wide deployment

> This release introduces cluster-wide deployment as a technology preview feature. A cluster-wide deployment contains a Service Mesh Control Plane that monitors resources for an entire cluster. The control plane uses a single query across all namespaces to monitor each Istio or Kubernetes resource kind that affects the mesh configuration. In contrast, the multitenant approach uses a query per namespace for each resource kind. Reducing the number of queries the control plane performs in a cluster-wide deployment improves performance.

Author: Done, thanks for the suggestion!

==== Configuring cluster-wide deployment

The following example `ServiceMeshControlPlane` object configures a cluster-wide deployment.

> To create an SMCP for cluster-wide deployment, a user must belong to the `cluster-admin` ClusterRole. If the SMCP is configured for cluster-wide deployment, it must be the only SMCP in the cluster". You cannot change the control plane mode from multitenant to cluster-wide (or from cluster-wide to multitenant). If a multitenant control plane already exists, delete it and create a new one.

Reviewer 4: ```suggestion
To create an SMCP for cluster-wide deployment, a user must belong to the `cluster-admin` ClusterRole. If the SMCP is configured for cluster-wide deployment, it must be the only SMCP in the cluster. You cannot change the control plane mode from multitenant to cluster-wide (or from cluster-wide to multitenant). If a multitenant control plane already exists, delete it and create a new one.
```

Author: change made

This example configures the SMCP for cluster-wide deployment.

[source,yaml]
----
  apiVersion: maistra.io/v2
  kind: ServiceMeshControlPlane
  metadata:
    name: cluster-wide
    namespace: istio-system
  spec:
    version: v2.3
    techPreview:
      controlPlaneMode: ClusterScoped <1>
----
<1> Enables Istiod to monitor resources at the cluster level rather than monitor each individual namespace.

> Additionally, the SMMR must also be configured for cluster-wide deployment. This example configures the SMMR for cluster-wide deployment.

Author: The doc uses SMCP and SMRR all over the place. It's common nomenclature and customers are vary familiar with it. I struggled with always wanting to write it out on every first instance when I was new to this project, but this is very common in the doc.

[source,yaml]
----
  apiVersion: maistra.io/v1
  kind: ServiceMeshMemberRoll
  metadata:
    name: default
  spec:
    members:
    - '*' <1>
----

> <1> Adds all namespaces to the mesh, including any namespaces you subsequently create. The following namespaces are not part of the mesh: kube, openshift, kube-* and openshift-*.

Author: Done, thanks for the suggestion

== New features {SMProductName} version {SMProductVersion}

This release of {SMProductName} introduces new features, addresses Common Vulnerabilities and Exposures (CVEs), contains bug fixes, and is supported on OpenShift Container Platform 4.9, 4.10, and 4.11.

=== Component versions included in {SMProductName} version {SMProductVersion}

|===
|Component |Version

|Istio
|1.14

|Envoy Proxy
|1.22.4

|Jaeger
|1.38

|Kiali
|1.57.3
|===

=== New Container Network Interface (CNI) DaemonSet container and configuration map

> The `openshift-operators` namespace includes a new istio CNI DaemonSet `istio-cni-node-v2-3` and a new configuration map `istio-cni-config-v2-3`.

Reviewer 4: I don't know if "configuration map" is how you usually refer to ConfigMaps in the documentation, but I think replacing "configuration map" with ConfigMap would make this sentence much clearer to the average OpenShift user.

Author: Edit made

When upgrading to Service Mesh Control Plane 2.3, the existing `istio-cni-node` DaemonSet is not changed, and a new `istio-cni-node-v2-3` DaemonSet is created.

> This namespace change does not affect previous releases or any `istio-cni-node` CNI DaemonSet associated with a Service Mesh Control Plane deployed using a previous release.

Reviewer 4: ```suggestion
This name change does not affect previous releases or any `istio-cni-node` CNI DaemonSet associated with a Service Mesh Control Plane deployed using a previous release.
```

Author: change made

=== Technology Preview
Some features in this release are currently in Technology Preview. These experimental features are not intended for production use.

[IMPORTANT]
====
Technology Preview features are not supported with Red Hat production service level agreements (SLAs) and might not be functionally complete. Red Hat does not recommend using them in production.
These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process. For more information about the support scope of Red Hat Technology Preview features, see the link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Support Scope].
====

==== OpenShift Service Mesh Console

This release introduces a Developer Preview version of the OpenShift Service Mesh Console, which integrates the Kiali interface directly into the OpenShift Console. For additional informaiton, see link:https://cloud.redhat.com/blog/introducing-the-openshift-service-mesh-console-a-developer-preview[Introducing the OpenShift Service Mesh Console (A Developer Preview)]

==== Cluster-Wide deployment

> This release introduces cluster-wide deployment as a technology preview feature. A cluster-wide deployment contains a Service Mesh Control Plane that monitors resources for an entire cluster. The control plane uses a single query across all namespaces to monitor each Istio or Kubernetes resource kind that affects the mesh configuration. Reducing the number of queries the control plane performs improves performance.

Reviewer 4: To explicitly call out the difference between the cluster-wide and the multi-tenant deployment, it might make sense to explain that the multi-tenant approach uses a query *per namespace* for each resource kind.

Author: Change made.

===== Configuring cluster-wide deployment

> The following example `ConfigMap` object configures a cluster-wide deployment.

Reviewer 4: ```suggestion
The following example `ServiceMeshControlPlane` object configures a cluster-wide deployment.
```

Author: change made

> There can only be one SMCP in the cluster. Creating the SMCP for cluster-wide deployment requires cluster administrator account privileges. You cannot change a control plane mode from multitenant to cluster-wide (or from cluster-wide to multitenant). If a multitenant control plane already exists, delete it and create a new one.

Reviewer 4: Maybe explain that "cluster administrator account privileges" means that the user must belong to the `cluster-admin` ClusterRole.

Reviewer 4: Also, "There can be only one SMCP in the cluster" is only true if that SMCP is configured as cluster-wide. Maybe something like "If the SMCP is configured for cluster-wide deployment, it must be the only SMCP in the cluster".

Author: changes made

[source,yaml]
----
  apiVersion: maistra.io/v2
  kind: ServiceMeshControlPlane
  metadata:
    name: cluster-wide
    namespace: istio-system
  spec:
    techPreview:
      controlPlaneMode: ClusterScoped <1>
    general:
      logging:
        componentLevels:

>           default: info

Reviewer 4: You can remove these lines.

Author: done

    version: v2.3
    runtime:
      defaults:
        container:
          imageRegistry: quay.io/luksa
          imageTag: latest
          imagePullSecrets:
          - name: quay-pull-secret
          resources:
            limits:
              cpu: 500m
              memory: 128Mi
      components:
        pilot:
          container:
            imageName: pilot
        wasmExtensions.cacher:
          container:
            imageName: pilot
        rateLimiting.rls:
          container:
            imageName: ratelimit
        prometheus:
          container:
            imageName: quay.io/maistra/prometheus-rhel8:latest-2.1-qe
    proxy:
      runtime:
        container:
          imageName: proxyv2
    ### end changes

    tracing:
      # change to Jaeger to enable tracing
      type: None
    addons:
      jaeger:
        name: jaeger
        install: {}
      grafana:
        enabled: false
        install: {}
      kiali:
        name: kiali
        enabled: true
        install: {}
      prometheus:
        enabled: true

>         install: {}

Reviewer 4: You should definitely remove most of these lines.

  ---
  apiVersion: maistra.io/v1
  kind: ServiceMeshMemberRoll
  metadata:
    name: default
  spec:
    members:

>     - '*' <2>

Reviewer 4: The ServiceMeshMemberRoll should probably be in a separate code block, with accompanying text explaining that using the asterisk in `spec.members` ensures that all namespaces are included in the mesh.

|===

=== `WasmPlugin` API

> This release adds support for the `WasmPlugin` API and deprecates the  `ServiceMeshExtension` API.

Reviewer 5: ```suggestion
This release adds support for the `WasmPlugin` API and deprecates the `ServiceMeshExtension` API.
```

Author: Done, thanks for the suggestion

----
Restricting route attachment on Gateway API listeners is possible using the `SameNamespace` or `All` settings. Istio ignores usage of label selectors in `listeners.allowedRoutes.namespaces` and reverts to the default behavior (`SameNamespace`).

== New features {SMProductName} 2.1.5.2

> This release of {SMProductName} addresses Common Vulnerabilities and Exposures (CVEs), bug fixes, and is supported on OpenShift Container Platform 4.9 or later.

Author: Done, thank you for the suggestion
